tests/test_events.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+import io
+import os
+import sys
+import json
+import unittest
+import logging
+import typing
+import boto3
+import time
+from uuid import uuid4
+from datetime import datetime
+
+from flashflood import replay_with_urls
+import requests
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+import dss
+from dss import events
+from dss.config import BucketConfig, Config, Replica, override_bucket_config
+from dss.util.version import datetime_to_version_format
+from dss.util import UrlBuilder
+from tests.infra import DSSAssertMixin, testmode
+from tests.infra.server import ThreadedLocalServer
+from tests import get_auth_header
+import tests
+
+
+logger = logging.getLogger(__name__)
+
+
+class TestEvents(unittest.TestCase, DSSAssertMixin):
+    @classmethod
+    def setUpClass(cls):
+        cls.app = ThreadedLocalServer()
+        cls.app.start()
+        cls.bundle = dict()
+        with override_bucket_config(dss.BucketConfig.TEST):
+            for replica in Replica:
+                bundle_uuid, bundle_version = cls._upload_bundle(replica)
+                cls.bundle[replica.name] = dict(uuid=bundle_uuid,
+                                                version=bundle_version,
+                                                key=f"bundles/{bundle_uuid}.{bundle_version}",)
+
+    @classmethod
+    def teardownClass(cls):
+        cls.app.shutdown()
+
+    def setUp(self):
+        Config.set_config(dss.BucketConfig.TEST)
+        events.record_event_for_bundle.cache_clear()
+        for replica in Replica:
+            name = f'DSS_{replica.name.upper()}_FLASHFLOOD_PREFIX'
+            os.environ[name] = f"flashflood-{uuid4()}"
+            events.record_event_for_bundle(replica, self.bundle[replica.name]['key'])
+
+    def test_list(self):
```

Comment:
Could this test a bad request as well as the working case?

Reply:
added test